### DISCUSSION TITLE
Design doc for Contact Us Feature

### DISCUSSION THREAD

**Discussion by TanishqSingla** [the-monkeys/the_monkeys#271] [General]
## Summary
Create a feature to enable users to send queries or report issues to site owners.

### Primary
- Create a contact us form to enable users to send us queries/reports
### Secondary (To be scoped)
- Create a issue tracking system, allowing admins to prioritize queries and create an interface for user to get updates.
## Implementation
**Frontend**
Contact us form with inputs
- Full name
	- Text input
	- Autofilled for logged in users
	- Required
- Email
	- Email input
	- Autofilled for logged in users
	- Required
- Query Type / Subject
	- Combo Box (Allowing custom input with presets in dropdown)
	- Required
- Attachments
	- Files with a limit size of 2MB
	- Optional
- CAPTCHA for bot prevention

**Backend**
- Endpoint for form submission
- Auto-reply features to notify users

**Comment by piyusharmap** [comment]
The design looks promising, but we can break it down into smaller parts and build it step by step.
